dashboard/modules/snapshot/snapshot_head.py

```diff
@@ -89,6 +106,47 @@ async def snapshot(self, req):
             success=True, message="hello", snapshot=snapshot
         )
 
+    @routes.get("/api/component_activities")
+    async def get_component_activities(self, req) -> aiohttp.web.Response:
+        # Get activity information for driver
+        timeout = req.query.get("timeout", None)
+        if timeout and timeout.isdigit():
+            timeout = int(timeout)
+        else:
+            timeout = 5
+
+        driver_activity_info = await self._get_job_activity_info(timeout=timeout)
+
+        resp = {"driver": dataclasses.asdict(driver_activity_info)}
+        return aiohttp.web.Response(
+            text=json.dumps(resp),
+            content_type="application/json",
+            status=aiohttp.web.HTTPOk.status_code,
+        )
+
+    async def _get_job_activity_info(self, timeout: int) -> RayActivityResponse:
+        # Returns if there is Ray activity from drivers (job).
+        # Drivers in namespaces that start with _ray_internal are not
+        # considered activity.
+        request = gcs_service_pb2.GetAllJobInfoRequest()
+        reply = await self._gcs_job_info_stub.GetAllJobInfo(request, timeout=timeout)
+
+        ray_internal_job_prefix = "_ray_internal"
```

Comment:
Can you use the constant here? (and change all other usage of this to a constant)